source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -398,6 +411,31 @@ void ConnectionManager::ActiveRpc::finalizeRequest() {
   }
 }
 
+bool ConnectionManager::ActiveRpc::passthroughSupported() {
+  if (passthrough_supported_.has_value()) {
+    return passthrough_supported_.value();
+  }
+
+  for (auto& entry : decoder_filters_) {
+    if (entry->handle_->passthroughSupported() == false) {
+      passthrough_supported_ = false;
+      return false;
+    }
+  }
+  passthrough_supported_ = true;
+  return true;
```

Comment:
How many times is this function called? Is there a lot of value in this global caching?

Reply:
It's called once in both request and response path. Will remove this caching.